specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2016-09-01-preview/searchservice.json

```diff
@@ -3957,6 +3986,111 @@
         }
       },
       "description": "Response from a List SynonymMaps request. If successful, it includes the full definitions of all synonym maps."
+    },
+    "ServiceStatistics": {
+      "properties": {
+        "counters": {
+          "$ref": "#/definitions/ServiceCounters",
+          "description": "Service level resource counters."
+        },
+        "limits": {
+          "$ref": "#/definitions/ServiceLimits",
+          "description": "Service level general limits."
+        }
+      },
+      "description": "Response from a get service statistics request. If successful, it includes service level counters and limits."
+    },
+    "ServiceCounters": {
+      "properties": {
+        "documentCount": {
+          "x-ms-client-name": "documentCounter",
+          "$ref": "#/definitions/ResourceCounter",
+          "description": "Total number of documents across all indexes in the service."
+        },
+        "indexesCount": {
+          "x-ms-client-name": "indexCounter",
+          "$ref": "#/definitions/ResourceCounter",
+          "description": "Total number of indexes."
+        },
+        "indexersCount": {
+          "x-ms-client-name": "indexerCounter",
+          "$ref": "#/definitions/ResourceCounter",
+          "description": "Total number of indexers."
+        },
+        "dataSourcesCount": {
+          "x-ms-client-name": "dataSourceCounter",
+          "$ref": "#/definitions/ResourceCounter",
+          "description": "Total number of data sources."
+        },
+        "storageSize": {
+          "x-ms-client-name": "storageSizeCounter",
+          "$ref": "#/definitions/ResourceCounter",
+          "description": "Total size of used storage in bytes."
+        },
+        "synonymMaps": {
+          "x-ms-client-name": "synonymMapCounter",
+          "$ref": "#/definitions/ResourceCounter",
+          "description": "Total number of synonym maps."
+        },
+      },
+      "description": "Represents a service level resource counters and quotas."
+    },
+    "ServiceLimits": {
+      "properties": {
+        "maxFieldsPerIndex": {
+          "type": "integer",
+          "format": "int32",
+          "x-nullable": true,
```

Comment:
All the limits could technically be null and while we could decide to prevent that by populating default values at the service side, I don't think we should.

Reply:
If all the limits could be null, then what about `maxIndexerRunTime`? Shouldn't it be `x-nullable` too for consistency?

Reply:
Are the limits null, or are there no limits? In general, if possible, it is preferable to not have semantically meaningful null values in your REST API. 

In this case, what would be the difference in meaning between a null value being sent back from the service, and no maxFieldsPerIndex attribute being included in the response?  If there isn't one, my suggestion to not send the key/value pair rather than sending a null value.

Reply:
*Limits* may be null if they don't apply, so I'm keeping them nullable, but I did take your suggestion for *counters* and removed the 'nullable' from resourceCounter.quota.

Reply:
@johanste We discussed offline and I agree with Nati that the intent here is to model limits that don't apply ("unlimited") as null, or a missing property. They both end up the same in the client programming model (null property). I don't know offhand whether our REST API will just omit properties with null values, but in either case changing that behavior now would affect our REST API globally, so it's not something we're likely to do.

Reply:
Slight correction; the client side programming model will depend on which SDK you are using (if any). The swagger document describes the REST API on the network layer. Anybody who does not use our SDKs (or who are using the "raw" layer  of the SDK - the one that doesn't deserialize responses into object models) will be exposed to the "value is null" vs. "key/value is missing" difference. Thus the general recommendation to stick to one way to express missing value (don't send key/value) unless there actually is a difference. 

If you could double-check what the service actually does (quite a few will suppress/not send null values) and have the swagger express that, it would be great.

Reply:
I have corrected the server side behavior just yesterday, to reflect the following changes:
For ‘Counters’ – ‘usage’ property could never be null. If a certain counter is not applicable (although we don’t have that notion in the real world) we won’t send the counter at all.
For ‘Limits’ – if a limit doesn’t exist\relevant, it’s value won’t return as null, but rather won’t be part of the response. Note that “null as unlimited” was never a thing for limits.  

These behaviors are already reflected by this PR, so no additional commits are required to realize them.